### PR TITLE
[C-553] Address sentry issues for 1.1.15

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Addresses top errors for 1.1.15 release:

https://sentry.io/organizations/audius/issues/3278477251/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3273253958/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3239725370/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3271472500/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/2990680632/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/2876540440/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3209192305/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3206125815/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3309918337/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3206340271/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

https://sentry.io/organizations/audius/issues/3306743063/?_allp=1&query=is%3Aunresolved+release.version%3A1.1.15&statsPeriod=14d

### How will this change be monitored?

Will check for regressions in next release

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-address-sentry-issues

File: packages/mobile/src/screens/collection-screen/CollectionScreen.tsx [hunk 1]
```diff
@@ -55,7 +55,8 @@ const useStyles = makeStyles(({ spacing }) => ({
 export const CollectionScreen = () => {
   const { params } = useRoute<'Collection'>()
 
-  const { id, searchCollection } = params
+  // params is incorrectly typed and can sometimes be undefined
+  const { id, searchCollection } = params ?? {}
```
Comment (raymondjacobson) [inline packages/mobile/src/screens/collection-screen/CollectionScreen.tsx:59]: just checking for sanity, but when these are undefined, we're guarded below on the !collection || !user check

File: packages/mobile/src/store/audio/reducer.ts [hunk 2]
```diff
@@ -108,13 +108,16 @@ const reducer = (
...
       // maybe not worth including sagas/thunks)
       if (playing) {
-        track(
-          make({
-            eventName: EventNames.PLAYBACK_PLAY,
-            id: `${state.queue[newIndex].trackId}`,
-            source: PlaybackSource.PASSIVE
-          })
-        )
+        const newTrack = state.queue[newIndex]
```
Comment (raymondjacobson) [inline packages/mobile/src/store/audio/reducer.ts:111]: just to i understand, this is probably when the user hits next on the end of a lineup?

File: packages/web/src/common/utils/uid.ts [hunk 3]
```diff
@@ -73,40 +87,47 @@ export function makeUids(kinds, ids, source) {
...
 
 /**
  * Gets the kind from a kind id
- * @param {String} kindId
+ * @param kindId
  */
-export const getKindFromKindId = kindId => kindId.split('-')[0]
+export const getKindFromKindId = (kindId: string) => kindId.split('-')[0]
 
 export const uuid = () => {
```
Comment (raymondjacobson) [inline packages/web/src/common/utils/uid.ts:116]: maybe we can just use a lib for this instead of typing our own

**Review by raymondjacobson — APPROVED** [review]

File: packages/mobile/src/screens/collection-screen/CollectionScreen.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/collection-screen/CollectionScreen.tsx:59]: yeah exactly, we just dont get user or collection and return null

File: packages/mobile/src/store/audio/reducer.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/audio/reducer.ts:111]: ohh that's smart, i was struggling to determine what was happening here, anything i should do different here given that context?

File: packages/web/src/common/utils/uid.ts (hunk 3 above)
Comment (dylanjeffers) [reply packages/web/src/common/utils/uid.ts:116]: def down for that, typing here helped me see all the callers that were making assumptions etc

File: packages/mobile/src/store/audio/reducer.ts (hunk 2 above)
Comment (raymondjacobson) [reply packages/mobile/src/store/audio/reducer.ts:111]: i think this is ok? we should test on web a decent bit just in case

**Pull request merged**